apps/kg/dataloader/KGDataset.py

```diff
@@ -98,14 +103,22 @@ def read_triple(self, path, mode, skip_first_line=False, format=[0,1,2]):
         tails = []
         rels = []
         with open(path) as f:
-            if skip_first_line:
-                _ = f.readline()
-            for line in f:
-                triple = line.strip().split('\t')
-                h, r, t = triple[format[0]], triple[format[1]], triple[format[2]]
-                heads.append(self.entity2id[h])
-                rels.append(self.relation2id[r])
-                tails.append(self.entity2id[t])
+            if partition == False:
+                if skip_first_line:
+                    _ = f.readline()
+                for line in f:
+                    triple = line.strip().split('\t')
+                    h, r, t = triple[format[0]], triple[format[1]], triple[format[2]]
+                    heads.append(self.entity2id[h])
+                    rels.append(self.relation2id[r])
+                    tails.append(self.entity2id[t])
+            else:
+                for line in f:
+                    triple = line.strip().split('\t')
+                    h, r, t = triple[format[0]], triple[format[1]], triple[format[2]]
+                    heads.append(int(h))
```

Comment:
It is wired to use self.entity2id[h] with partition == False with use int[h] with partition == True. It is confusing here. Actually, when we provide the partitioned data, it has already been preprocessed.
Can we create a ParitionKGDataset based on KGDataset?

Reply:
Sure. let me create a PartitionKGDataset.